### PR TITLE
feature: Address lookup

### PR DESCRIPTION
**Pull request by goergisn**
## Changes

<new>

* Adding functionality to provide billing address completion via a lookup handler which is called when the user searches for an address.
* You can use address lookup by setting the `configuration.card.billingAddress.mode` to `.lookup(handler:)`

</new>

https://github.com/Adyen/adyen-ios/assets/4838877/15ea5c5e-f50e-431f-86fc-96432b32aa0b

### CODE REVIEW

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController.swift [hunk 1]
```diff
@@ -35,9 +35,38 @@ internal class AddressLookupSearchViewController: SearchViewController {
...
+                listItemStyle.title.color = .Adyen.componentSecondaryLabel
+                listItemStyle.title.textAlignment = .center
+                listItemStyle.subtitle.font = .preferredFont(forTextStyle: .subheadline)
+                listItemStyle.subtitle.color = .Adyen.defaultBlue
+                listItemStyle.subtitle.textAlignment = .center
+                
+                itemsWithManualButton.append(
+                    .init(
+                        title: "Select your address or use",
+                        subtitle: "manual address entry",
```
Comment (descorp) [inline Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController.swift]: Can it be a single title?

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+Style.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import UIKit
+
+public struct AddressLookupSearchStyle: ViewStyle {
```
Comment (erenbesel) [inline Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+Style.swift:10]: public type missing documentation

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+EmptyView.swift [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            fatalError("init(coder:) has not been implemented")
+        }
+    }
+}
+
+// MARK: - Updating Interface
+
+private extension AddressLookupSearchViewController.EmptyView {
+    
+    func setupContent() {
```
Comment (erenbesel) [inline Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+EmptyView.swift]: small thing but this one is identical to search controllers `setupContent()`. Was thinking if we should make a common protocol/func for it

Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+EmptyView.swift]: Great catch! The views are actually so similar that I'll create a generic version of it 👌

Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+EmptyView.swift]: Fixed in 1f2519998a83402396d39b279bf7e24a717a8213

File: Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+Style.swift (hunk 2 above)
Comment (goergisn) [reply Adyen/UI/View Controllers/AddressLookup/Search/AddressLookupSearchViewController+Style.swift:10]: Fixed in: 1f2519998a83402396d39b279bf7e24a717a8213

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Pull request merged**